src/core/dynamic-user.c

```diff
@@ -628,6 +628,11 @@ int dynamic_user_serialize_one(DynamicUser *d, const char *key, FILE *f, FDSet *
         if (copy1 < 0)
                 return log_error_errno(copy1, "Failed to add dynamic user storage fd to serialization: %m");
 
+        if (store_index) {
+                copy0 = fdset_size(fds) - 2;
+                copy1 = copy0 + 1;
+        }
+
         (void) serialize_item_format(f, key, "%s %i %i", d->name, copy0, copy1);
```

Comment:
i'd prefer if we'd come to a distinct serialiazation for this.

maybe suffix "-by-fd-index" to the key string, or by prefixing the fds individually by a `#` or so in the output? i think the latter is a bit nice.

i.e. 


when serializing indexes:
```
myfditem waldo #4711 #4712
```
when serializing fds:
```
myfditem waldo 23 27
```

you get the idea...

Reply:
I went with the first option, as to me it looks better and simpler, and doesn't complicate parsing the value